internal/provider/resource_workspace_participant.go

```diff
@@ -0,0 +1,140 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/healx/terraform-provider-nftower/internal/client"
+)
+
+func resourceWorkspaceParticipant() *schema.Resource {
+	return &schema.Resource{
+		Description: "Grants access to a tower workspace.",
+
+		CreateContext: resourceWorkspaceParticipantCreate,
+		ReadContext:   resourceWorkspaceParticipantRead,
+		UpdateContext: resourceWorkspaceParticipantUpdate,
+		DeleteContext: resourceWorkspaceParticipantDelete,
+
+		Schema: map[string]*schema.Schema{
+			"workspace_id": {
+				Description: "The id of the workspace to grant access to.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"member_id": {
+				Description: "The id of the member in the organization.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"role": {
+				Description: "The role of the participant.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "launch",
```

Comment:
Could also be "view" as a "default read-only" kind of role. Naturally, it would be overridden when used